cherrypy/_cpreqbody.py

```diff
@@ -470,6 +470,10 @@ def __init__(self, fp, headers, params=None, parts=None):
                     self.filename.endswith('"')
                 ):
                     self.filename = self.filename[1:-1]
+            elif 'filename*' in disp.params:
+                # @see https://tools.ietf.org/html/rfc5987
+                fn_encoding, _, filename = disp.params['filename*'].split("'")
+                self.filename = unquote_qs(filename, fn_encoding)
```

Comment:
Is `unquote_qs` the right operation here? I'm skeptical because it translates `+` to ` `, which I don't see in the spec. That's also a bug with the existing _cp_compat; it's doing too much already. Let's fix that first.

Reply:
So I can't remember exactly but I think the way I tested it on some older version was I POSTed a file with spaces and some UTF8 chars using `requests` and saw that `requests` is using quote.

Reply:
`requests` is often not a good example of following RFCs, unfortunately.
As for `+` vs `%20` for ` `, I guess it depends on how client decides to encode that. I think I saw something regarding this in query params spec in some rfc.